File: core/src/main/java/org/treblereel/gwt/three4g/loaders/Cache.java
package org.treblereel.gwt.three4g.loaders;

import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsType;

@JsType(isNative = true, name = "THREE.Cache", namespace = JsPackage.GLOBAL)
public class Cache {
  public static boolean enabled;
  public static double files;

  public static native void add(String key, double file);

  public static native void clear();

  public static native double get(String key);

  public static native void remove(String key);
}

File: core/src/main/resources/org/treblereel/gwt/three4g/loaders/Cache.js
/** @type {Object<string, Object>} */
const files = {};

class _Cache {

	constructor() {

		this.enabled = false;

		/** @type {Object<string, Object>} */
		this.files;

	}

	/**
	 * 
	 * @param {string} key 
	 * @param {Object} file 
	 */
	add( key, file ) {

		if ( this.enabled === false ) return;

		files[ key ] = file;

	}

	/**
	 * 
	 * @param {string} key 
	 * @return {Object|undefined} 
	 */
	get( key ) {

		if ( this.enabled === false ) return;

		return files[ key ];

	}

	/**
	 * 
	 * @param {string} key 
	 */
	remove( key ) {

		delete files[ key ];

	}

	clear() {

		this.files = {};

	}

}





const Cache = new _Cache();


export { Cache };